modules/starboard/handleStarboard.js

```diff
@@ -0,0 +1,64 @@
+const { embedTypeV2, disableModule, formatDiscordUserName } = require('../../src/functions/helpers');
+const { localize } = require('../../src/functions/localize');
+
+module.exports = async (client, msgReaction, isReactionRemove = false) => {
+    if (!client.botReadyAt) return;
+    const msg = msgReaction.message;
+    if (!msg.guild) return;
+    if (msg.guild.id !== client.guildID) return;
+    if (msgReaction.partial) msgReaction = await msgReaction.fetch();
+
+    const starConfig = client.configurations['starboard']['config'];
+    if (!starConfig || (starConfig.emoji !== msgReaction.emoji.name && starConfig.emoji !== msgReaction.emoji.id)) return;
+    if (isNaN(starConfig.minStars)) return disableModule('starboard', localize('starboard', 'invalid-minstars', {stars: starConfig.minStars}));
+
+    const starMsg = await client.models['starboard']['StarMsg'].findOne({
+        where: {
+            msgId: msg.id
+        }
+    });
+
+    const channel = client.channels.cache.get(starConfig.channelId);
+    if (!channel) return disableModule('starboard', localize('partner-list', 'channel-not-found', {c: starConfig.channelId}));
+    if ((msg.channel.nsfw && !channel.nsfw) || starConfig.excludedChannels.includes(msg.channel.id) || starConfig.excludedRoles.some(r => msg.member.roles.cache.has(r))) return;
+
+    let reactioncount = msgReaction.count;
+    if (!starConfig.selfStar && msgReaction.users.cache.has(msg.author.id)) reactioncount--;
+
+    const starboardMsg = starMsg ? await channel.messages.fetch(starMsg.starMsg).catch(() => {}) : null;
+    if (reactioncount < starConfig.minStars) {
+        if (isReactionRemove) {
+            if (starboardMsg) starboardMsg.delete();
+            client.models['starboard']['StarMsg'].destroy({
+                where: {
+                    msgId: msg.id
+                }
+            });
+        }
+        return;
+    }
+
+    const generatedMsg = await embedTypeV2(starConfig.message, {
+        '%stars%': msgReaction.count,
+        '%content%': msg.content,
+        '%username%': msg.author.username,
+        '%usertag%': formatDiscordUserName(msg.author),
+        '%nickname%': msg.member ? msg.member.displayName : msg.author.username,
```

Comment:
In contrast to `.nickname`, `.displayName` will automatically default to msg.author.username[^1].

[^1]: https://github.com/discordjs/discord.js/blob/759c0b0d46ce67f4208cdf22214d2272860f27b9/src/structures/GuildMember.js#L267